pkg/core/native/policy.go

```diff
@@ -306,32 +301,26 @@ func (p *Policy) GetMaxBlockSystemFeeInternal(dao dao.DAO) int64 {
 	return p.getInt64WithKey(dao, maxBlockSystemFeeKey)
 }
 
-// getBlockedAccounts is Policy contract method and returns list of blocked
-// accounts hashes.
-func (p *Policy) getBlockedAccounts(ic *interop.Context, _ []stackitem.Item) stackitem.Item {
-	ba, err := p.GetBlockedAccountsInternal(ic.DAO)
-	if err != nil {
-		panic(err)
-	}
-	return ba.ToStackItem()
+// isBlocked is Policy contract method and checks whether provided account is blocked.
+func (p *Policy) isBlocked(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	hash := toUint160(args[0])
+	return stackitem.NewBool(p.IsBlockedInternal(ic.DAO, hash))
 }
 
-// GetBlockedAccountsInternal returns list of blocked accounts hashes.
-func (p *Policy) GetBlockedAccountsInternal(dao dao.DAO) (BlockedAccounts, error) {
+// IsBlockedInternal checks whether provided account is blocked
+func (p *Policy) IsBlockedInternal(dao dao.DAO, hash util.Uint160) bool {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 	if p.isValid {
-		return p.blockedAccounts, nil
+		_, ok := p.blockedAccounts[hash]
```

Comment:
`return p.blockedAccounts[hash]`?

Reply:
It's array now, so replaced with `range`.